cirq-core/cirq/devices/grid_device_metadata.py

```diff
@@ -146,27 +159,46 @@ def _value_equality_values_(self):
             self._gateset,
             tuple(duration_equality),
             tuple(sorted(self.qubit_set)),
+            frozenset(self._compilation_target_gatesets),
         )
 
     def __repr__(self) -> str:
         return (
             f'cirq.GridDeviceMetadata({repr(self._qubit_pairs)},'
             f' {repr(self._gateset)}, {repr(self._gate_durations)},'
-            f' {repr(self.qubit_set)})'
+            f' {repr(self.qubit_set)}, {repr(self._compilation_target_gatesets)})'
         )
 
     def _json_dict_(self):
         duration_payload = None
         if self._gate_durations is not None:
             duration_payload = sorted(self._gate_durations.items(), key=lambda x: repr(x[0]))
 
-        return {
+        jd = {
             'qubit_pairs': sorted(list(self._qubit_pairs)),
             'gateset': self._gateset,
             'gate_durations': duration_payload,
             'all_qubits': sorted(list(self.qubit_set)),
         }
+        if len(self._compilation_target_gatesets) > 0:
+            jd['compilation_target_gatesets'] = list(self._compilation_target_gatesets)
```

Comment:
I don't think we need the conditional logic here. It's okay to append serialization until the deserialization is still supported.